server/controllers/UsersController.js

```diff
@@ -98,6 +100,132 @@ class UsersController {
         }
       }));
   }
+
+  /**
+ * @description - Route handler to send password reset link
+ * @param {object} req - request object
+ * @param {object} res - response object
+ * @returns {object} - returns object
+ */
+  static forgotPassword(req, res) {
+    const { email } = req.body;
+    const lifeSpan = '1h'; // token expires after 1 hour
+
+    User.findOne({
+      where: {
+        email
+      }
+    })
+      .then((userFound) => {
+        if (!userFound) {
+          return res.status(404).json({
+            errors: {
+              message: ['No account with that email address exists']
+            }
+          });
+        }
+
+        const userId = userFound.id;
+        const token = createToken(userId, lifeSpan);
+        const forgotUrl = `${req.protocol}://${req.host}/password_reset`
+        + `/${token}`;
+
+        const user = {
+          id: userId,
+          email
+        };
+
+        const message = forgotPasswordEmail(forgotUrl);
+
+        // send email
+        sendEmail(user, message);
+
+        res.status(200).send({
+          status: 'success',
+          message: ['Password reset email sent successfully']
+        });
+      })
+      .catch((err) => {
+        res.status(500)
+          .json({
+            error: {
+              message: err.message,
+            },
+          });
+      });
+  }
+
+  /**
+   * @description - Route handler to reset user password
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @returns {object} - returns object
+   */
+  static resetPassword(req, res) {
+    const { password, confirmPassword } = req.body;
+    const { token } = req.params;
+
+    if (password === confirmPassword) {
+      jwt.verify(token, process.env.JWT_SECRET, (err, decoded) => {
+        if (err) {
+          return res.status(400).send({
+            status: 'failure',
+            message: ['Password reset token is invalid or has expired']
+          });
+        }
+
+        const userId = decoded.id;
+        User.findOne({
+          where: {
+            id: userId
+          },
+          attributes: ['id', 'fullName', 'email', 'avatarUrl', 'bio']
+        })
+          .then((user) => {
+            if (!user) {
+              return res.status(404).send({
+                status: 'failure',
+                message: ['User not found'],
+              });
+            }
+
+            const hashedPassword = bcrypt.hashSync(`${password}`);
+
+            user.update({ password: hashedPassword })
+              .then(() => {
+                const message = resetPasswordEmail();
+
+                // send email
+                sendEmail(user, message);
+
+                res.status(200).send({
```

Comment:
When the user resets password successfully, I'm not sure we want to return all of his info. BTW, are we assuming that once this happens he is auto signed in? If so, are we generating an auth token for this?

Reply:
You are right about the bio, I will take that out. My idea for what happens after a password reset was for the user to get routed to the login form.

Reply:
If so, don't we need to generate a login token for them?

Reply:
I don't think so, the user should get a token after they have logged in.